### ISSUE TITLE
Support regular Nostr filters

### ISSUE THREAD

**Issue by alexgleason** [PrimalHQ/primal-caching-service#13]
If Primal's caching service could support regular Nostr filters, it would greatly increase its value. It would appeal to self-hosters who would install this as part of self-hosted Nostr infrastructure. The current filters work only with Primal apps.

**Comment by pritk** [comment]
you are right. most of currently supported filters in Primal caching service is specific to apps like existing Primal apps. but we can already imagine different nostr clients that can benefit from existing cache server API. otoh we decided to use one of the best software for relay functionality in our system. so we are looking at this as combination of the best options for each aspect of the system (relaying: strfry, caching: primal caching service).

**Comment by alexgleason** [comment]
How about search? I can see the cache service accepts this: `{"cache":["user_search",{"query":"alex","limit":10}]}`

It could easily support [NIP-50](https://github.com/nostr-protocol/nips/blob/master/50.md) search, making it usable by clients as a dedicated Search relay.
